ocaml/xapi/xapi_vm.ml

```diff
@@ -195,6 +195,9 @@ let start ~__context ~vm ~start_paused ~force =
 		debug "Setting ha_always_run on vm=%s as true during VM.start" (Ref.string_of vm)
 	end;
 
+	(* Check to see if we're using any restricted platform kvs. This raises
+	   an exception if so *)
+	Xapi_vm_helpers.check_restricted_platform_flags ~__context vmr.API.vM_platform;
```

Comment:
Isn't this better done in Xapi_xenopsd when parsing the platform map? You would also handle the VM.resume case that way. And there is already some code there that manipulates the platform map.